libs/traits/src/rewards.rs

```diff
@@ -182,6 +182,24 @@ pub trait CurrencyGroupChange {
 	fn currency_group(currency_id: Self::CurrencyId) -> Option<Self::GroupId>;
 }
 
+pub trait RewardIssuance {
+	/// Type used to identify the beneficiary.
+	type AccountId;
+
+	/// Type used to identify the currency
+	type CurrencyId;
+
+	/// Type used as balance for all currencies and reward.
+	type Balance;
+
+	/// Issue the provided reward amount to a beneficiary account address.
+	fn issue_reward(
```

Comment:
I'm thinking that all implementers of these traits will call `issue_reward()` in all implementations of `reward_group()`. Could we extract this logic and make this call automatic?. For example, in line 110 of this same file, where `reward_group()` is called, we can use the returned balance to call `issue_reward()` automatically just in the next line. This way the implementater no longer needs to care to call this explicitely. WDYT?

Reply:
I am a fan of the idea, however it blows up the code quite significantly as we would have to introduce 
* `RewardCurrency`
* Truncated pallet account id
* `RewardIssuance`

to the `RewardMechanism` trait or `RewardMechanism::reward_group` call. Wanted to re-check with you before I start the endeavour. Or am I missing something here?

Reply:
I'm not seeing why it requires to modify the reward mechanism, could you expand? 👀 

As I see this is adding the bound of `trait DistributedRewards: GroupRewards + RewardIssuance`, and then, in the default implementation of `distribute_reward_with_weights()`.

```rust
fn distribute_reward_with_weights() {
    // ...
    let (account, currency, balance) = Self::reward_group(group_id, group_reward);
    RewardIssuance::issue_reward(account, currency, balance);
    balance
    // ...
}
```

It can add more "code" in the sense that `reward_group()` should return the account and the currency too. But it should be the `reward_group()` from the pallet, not from the `RewardMechanism`.

Reply:
Rethinking this, maybe instead of 
```rust 
trait DistributedRewards: GroupRewards + RewardIssuance
```
is better:
```rust
fn distribute_reward_with_weights<RewardIssuance>()
```

Reply:
> It can add more "code" in the sense that reward_group() should return the account and the currency too. But it should be the reward_group() from the pallet, not from the RewardMechanism.

1. This was one of my points: How does `reward_group()` have knowledge of the recipient account and reward currency if we don't add this to the input arguments?

> I'm not seeing why it requires to modify the reward mechanism, could you expand? 👀

2.  In #1998, I haven't used `distribute_reward_with_weights` yet but `reward_group` instead, as we only have a single group. It appears to me this not desired, e.g. `distribute_reward_with_weights` should be called for the distribution as the name suggests?

Reply:
> It can add more "code" in the sense that reward_group() should return the account and the currency too. But it should be the reward_group() from the pallet, not from the RewardMechanism.

1. This was one of my points: How does `reward_group()` have knowledge of the recipient account and reward currency if we don't add this to the input arguments?

> I'm not seeing why it requires to modify the reward mechanism, could you expand? 👀

2.  In #1998, I haven't used `distribute_reward_with_weights` yet but `reward_group` instead, as we only have a single group. It appears to me this not desired, e.g. `distribute_reward_with_weights` should be called for the distribution as the name suggests?

Reply:
> I haven't used distribute_reward_with_weights yet but reward_group instead

Ok, I think this is the main point. If `distribute_reward_with_weights` (or `distribute_reward`) is not used, all that I said lose sense 😅 . I always called to `distribute_reward` directly and I forgot it wasn't the same case here. 🙏🏻

Reply:
It appears to me that we want pallets which use `pallet-rewards` to actually call `distribute_reward` instead of `reward_group` right? Then I would go forward with refactoring to stay consistent.

Reply:
> It appears to me that we want pallets which use pallet-rewards to actually call distribute_reward instead of reward_group right?

Yes, that was at least what I had in mind. But it's true that your use case can bypass the `distribute_reward()` call without damage. For me, it's ok as it is 👍🏻 . Maybe we can change this in the future if `distribute_reward() has more relevance.